SpatialGDK/Source/SpatialGDK/Public/Utils/InspectionColours.h

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Containers/UnrealString.h"
+#include "Math/Color.h"
+#include "Math/UnrealMathUtility.h"
+#include "Misc/Char.h"
+
+// Mimicking Inspector V2 colouring from platform/js/console/src/inspector-v2/styles/colors.ts
+
+namespace SpatialGDK
+{
+	namespace
+	{
+		const int32 MIN_HUE = 10;
+		const int32 MAX_HUE = 350;
+		const int32 MIN_SATURATION = 60;
+		const int32 MAX_SATURATION = 100;
+		const int32 MIN_LIGHTNESS = 25;
+		const int32 MAX_LIGHTNESS = 60;
+
+		double GenerateValueFromThresholds(int64 Hash, int32 Min, int32 Max)
+		{
+			return Hash % FMath::Abs(Max - Min) + Min;
+		}
+
+		FColor HLAtoRGB(double Hue, double Saturation, double Lightness)
```

Comment:
Why not HSLtoRGB, idk what HLA is, normally it would be HSLA but there is no alpha involved here as far as I can see. For reference the equivalent function in the TS file you linked is called hslToRgb.

Reply:
I think my brain is stuck in draco world with HLA (high-level architecture) 😂